src/allotropy/parsers/thermo_fisher_nanodrop_eight/nanodrop_eight_parser.py

```diff
@@ -0,0 +1,272 @@
+        contents, filename = named_file_contents
+        data = NanoDropEightReader.read(contents)
+        data = self._add_measurement_uuids(data)
+        return self._get_model(data, filename)
+
+    def _get_model(self, data: pd.DataFrame, filename: str) -> Model:
+        return Model(
+            field_asm_manifest="http://purl.allotrope.org/manifests/spectrophotometry/BENCHLING/2023/12/spectrophotometry.manifest",
+            spectrophotometry_aggregate_document=SpectrophotometryAggregateDocument(
+                spectrophotometry_document=self._get_spectrophotometry_document(data),
+                calculated_data_aggregate_document=CalculatedDataAggregateDocument(
+                    calculated_data_document=self._get_calculated_data_document(data),
+                ),
+                data_system_document=DataSystemDocument(
+                    file_name=filename,
+                    ASM_converter_name=ASM_CONVERTER_NAME,
+                    ASM_converter_version=ASM_CONVERTER_VERSION,
+                ),
+                device_system_document=DeviceSystemDocument(
+                    model_number="Nanodrop Eight",
+                    device_identifier="Nanodrop",
+                ),
+            ),
+        )
+
+    def _add_measurement_uuids(self, data: pd.DataFrame) -> pd.DataFrame:
+        data["A260 uuid"] = [str(uuid.uuid4()) for _ in range(len(data.index))]
+        data["A280 uuid"] = [str(uuid.uuid4()) for _ in range(len(data.index))]
+        return data
+
+    def _get_spectrophotometry_document(
+        self, data: pd.DataFrame
+    ) -> list[SpectrophotometryDocumentItem]:
+        return [
+            self._get_spectrophotometry_document_item(data, i)
+            for i in range(len(data.index))
+        ]
+
+    def _get_calculated_data_document(
+        self, data: pd.DataFrame
+    ) -> list[CalculatedDataDocumentItem]:
+        calculated_data_documents = []
+        for i in range(len(data.index)):
+            if _get_str(data, i, "260/280"):
+                calculated_data_documents.append(self._get_260_280(data, i))
+
+            if _get_str(data, i, "260/230"):
+                calculated_data_documents.append(self._get_260_230(data, i))
+
+        return calculated_data_documents
+
+    def _get_260_280(self, data: pd.DataFrame, row: int) -> CalculatedDataDocumentItem:
+
+        return CalculatedDataDocumentItem(
+            calculated_data_name="A260/280",
+            calculated_result=TQuantityValue(
+                value=_get_float(data, row, "260/280"), unit="unitless"
+            ),
+            calculated_data_identifier=str(uuid.uuid4()),
+            data_source_aggregate_document=DataSourceAggregateDocument1(
+                [
+                    DataSourceDocumentItem(
+                        data_source_feature="absorbance",
+                        data_source_identifier=str(_get_str(data, row, "A260 uuid")),
```

Comment:
I'm skeptical of every instance of `str(_get_str(`, both from a naming perspective and looking at the impl.

Calling `str` on `None` returns `'None'`, which I really don't think is what we want.

For fields like `data_source_identifier` that require a `str`, we need to make sure we supply one. It seems like it should always exist since the columns were added though.